src/Microsoft.VisualStudio.Services.Agent/ProcessInvoker.cs

```diff
@@ -213,6 +222,13 @@ public sealed class ProcessInvoker : AgentService, IProcessInvoker
                 _proc.StandardInput.Dispose();
             }
 
+            if(!waitForExit)
+            {
+                //just give some time for the AgentService.exe to warm up
+                Thread.Sleep(5*1000);
```

Comment:
>Thread.Sleep(5*1000); [](start = 16, length = 21)

if we are not waiting for exit then we should not wait at all. what is the reason for this sleep